expression/constant.go

```diff
@@ -118,9 +150,45 @@ func (c *Constant) Eval(_ chunk.Row) (types.Datum, error) {
 	return c.Value, nil
 }
 
+// UnsafeEval returns datum maybe not match const's type.
```

Comment:
Is that only when `c.ParamMarker` is not `nil` the type of returned `Datum` may not match this `Constant`?